cassovary-core/src/main/scala/com/twitter/cassovary/graph/Node.scala

```diff
@@ -273,9 +289,15 @@ trait Node {
 
 object Node {
   private lazy val randGen: Random = new Random
-  def apply(nodeId: Int, out: Seq[Int], in: Seq[Int]) = new Node {
-    val id = nodeId
-    def inboundNodes() = in
-    def outboundNodes() = out
+
+  def apply(nodeId: Int, out: Seq[Int], in: Seq[Int]): Node = {
+    new SeqBasedNode(nodeId, in, out)
+  }
+
+  def withSortedNeighbors(nodeId: Int, in: Array[Int], out: Array[Int]) = {
+    new SeqBasedNode(nodeId, in, out) with SortedNeighborsNodeOps
   }
 }
+
+class SeqBasedNode private[graph] (val id: Int, val inboundNodes: Seq[Int], val outboundNodes: Seq[Int])
```

Comment:
My question was different -- as currently defined, the purpose of SeqBasedNode is not clear at all (to me). Seems like it doesn't provide anything new or different than Node itself. i.e., whatever you do with SeqBasedNode you can directly do based on Node.